### ISSUE TITLE
bug report: receiving address checked against LDAP 

### ISSUE THREAD

**Issue by ToxicMushroom** [docker-mailserver/docker-mailserver#3538]
### 📝 Preliminary Checks

- [X] I tried searching for an existing issue and followed the [debugging docs](https://docker-mailserver.github.io/docker-mailserver/latest/config/debugging/) advice, but still need assistance.

### 👀 What Happened?

Today I converted my docker-mailserver to use LDAP as authentication mechanism.
Receiving mails works well, from outside (e.g. @gmail.com and inside @melijn.com).
Sending email between internal users (@melijn.com) works.
**Sending emails to mailboxes outside (e.g. @gmail.com does not).**

I did not see any mention of configuration to which addresses you can send to unless I misunderstood the docs: https://docker-mailserver.github.io/docker-mailserver/latest/config/advanced/auth-ldap/.

### 👟 Reproduction Steps

Env: [mail-server.env.txt](https://github.com/docker-mailserver/docker-mailserver/files/12622664/mail-server.env.txt)

LDAP Search query: 
```
I have no name!@d0d034e122f5:/$ ldapsearch -x -H ldaps://ldap.melijn.com -b 'dc=melijn,dc=com' -D "cn=admin,dc=melijn,dc=com" -W "(&(objectClass=inetOrgPerson)(mail=merlijn@melijn.com))"
```
```
# extended LDIF
#
# LDAPv3
# base <dc=melijn,dc=com> with scope subtree
# filter: (&(objectClass=inetOrgPerson)(mail=merlijn@melijn.com))
# requesting: ALL
#

# merlijn, users, melijn.com
dn: cn=merlijn,ou=users,dc=melijn,dc=com
givenName: redacted1
sn: redacted2
cn: merlijn
uid: merlijn
uidNumber: 1002
gidNumber: 500
homeDirectory: /home/users/merlijn
objectClass: inetOrgPerson
objectClass: posixAccount
objectClass: top
mail: merlijn@melijn.com
userPassword:: redacted

# search result
search: 2
result: 0 Success

# numResponses: 2
# numEntries: 1
```

### 🐋 DMS Version

v12.1.0

### 💻 Operating System and Architecture

5.4.0-162-generic #179-Ubuntu

### ⚙️ Container configuration files

```yml
version: '3.7'

services:
  mail:
    image: mailserver/docker-mailserver:latest
    hostname: mail
    domainname: melijn.com
    container_name: mail
    ports:
      - "25:25"
      - "143:143"
      - "587:587"
      - "993:993"
    volumes:
      - maildata:/var/mail
      - mailstate:/var/mail-state
      - maillogs:/var/log/mail
      - ./config/:/tmp/docker-mailserver/
      - /opt/traefik/acme.json:/etc/letsencrypt/acme.json:ro
    env_file:
      - .env
      - mail-server.env
    networks:
      - proxy_default
    cap_add:
      - NET_ADMIN
      - SYS_PTRACE
    restart: always

  nginx2:
    image: nginx:latest
    container_name: "gen-nginx-mail-cert"
    volumes:
      - /data/www/:/usr/share/nginx/html/
    networks:
      - proxy_default
    labels:
      - "traefik.http.routers.certgen.rule=Host(`mail.melijn.com`)"
      - "traefik.enable=true"
      - "traefik.http.services.certgen.loadbalancer.server.port=80"
      - "traefik.http.routers.certgen.tls=true"
      - "traefik.http.routers.certgen.entrypoints=websecure"
      - "traefik.http.routers.certgen.tls.certresolver=le"

networks:
  proxy_default:
    external: true

volumes:
  maildata:
  mailstate:
  maillogs:
```

### 📜 Relevant log output

```Text
mail                 | Sep 15 18:35:25 mail postfix/submission/smtpd[1005]: connect from 113.168-65-87.adsl-dyn.isp.belgacom.be[87.65.168.113]
mail                 | Sep 15 18:35:26 mail postfix/submission/smtpd[1005]: Anonymous TLS connection established from 113.168-65-87.adsl-dyn.isp.belgacom.be[87.65.168.113]: TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits) key-exchange X25519 server-signature RSA-PSS (4096 bits) server-digest SHA256
mail                 | Sep 15 18:35:26 mail dovecot: imap-login: Login: user=<merlijn>, method=PLAIN, rip=127.0.0.1, lip=127.0.0.1, mpid=1010, secured, session=<dTRgDmoFwIN/AAAB>
mail                 | Sep 15 18:35:26 mail dovecot: imap(merlijn)<1010><dTRgDmoFwIN/AAAB>: Disconnected: Logged out in=16 out=496 deleted=0 expunged=0 trashed=0 hdr_count=0 hdr_bytes=0 body_count=0 body_bytes=0
mail                 | Sep 15 18:35:26 mail postfix/submission/smtpd[1005]: 3F8FA227F4: client=113.168-65-87.adsl-dyn.isp.belgacom.be[87.65.168.113], sasl_method=PLAIN, sasl_username=merlijn@melijn.com
mail                 | Sep 15 18:35:26 mail postfix/sender-cleanup/cleanup[1012]: 3F8FA227F4: message-id=<a1351c88-8e7b-4fb6-bca1-53b455f05475@melijn.com>
mail                 | Sep 15 18:35:26 mail postfix/sender-cleanup/cleanup[1012]: 3F8FA227F4: replace: header MIME-Version: 1.0 from 113.168-65-87.adsl-dyn.isp.belgacom.be[87.65.168.113]; from=<merlijn@melijn.com> to=<pixelhamsteryt@gmail.com> proto=ESMTP helo=<[192.168.1.33]>: MIME-Version: 1.0
mail                 | Sep 15 18:35:26 mail opendkim[873]: 3F8FA227F4: DKIM-Signature field added (s=mail, d=melijn.com)
mail                 | Sep 15 18:35:26 mail postfix/qmgr[985]: 3F8FA227F4: from=<merlijn@melijn.com>, size=374, nrcpt=1 (queue active)
mail                 | Sep 15 18:35:26 mail dovecot: imap-login: Login: user=<merlijn>, method=PLAIN, rip=87.65.168.113, lip=172.18.0.25, mpid=1016, TLS, session=<NQJnDmoFDLJXQahx>
mail                 | Sep 15 18:35:26 mail dovecot: imap(merlijn)<1016><NQJnDmoFDLJXQahx>: Disconnected: Logged out in=464 out=624 deleted=0 expunged=0 trashed=0 hdr_count=0 hdr_bytes=0 body_count=0 body_bytes=0
mail                 | Sep 15 18:35:27 mail postfix/smtpd-amavis/smtpd[1018]: connect from localhost[127.0.0.1]
mail                 | Sep 15 18:35:27 mail postfix/smtpd-amavis/smtpd[1018]: E33D722F63: client=localhost[127.0.0.1]
mail                 | Sep 15 18:35:27 mail postfix/cleanup[1019]: E33D722F63: message-id=<a1351c88-8e7b-4fb6-bca1-53b455f05475@melijn.com>
mail                 | Sep 15 18:35:27 mail postfix/qmgr[985]: E33D722F63: from=<merlijn@melijn.com>, size=1083, nrcpt=1 (queue active)
mail                 | Sep 15 18:35:27 mail amavis[1003]: (01003-01) Passed CLEAN {RelayedOpenRelay}, [87.65.168.113]:54144 [87.65.168.113] <merlijn@melijn.com> -> <pixelhamsteryt@gmail.com>, Queue-ID: 3F8FA227F4, Message-ID: <a1351c88-8e7b-4fb6-bca1-53b455f05475@melijn.com>, mail_id: h6HUMdGGE7aH, Hits: -0.202, size: 871, queued_as: E33D722F63, 1602 ms
mail                 | Sep 15 18:35:27 mail postfix/smtp-amavis/smtp[1013]: 3F8FA227F4: to=<pixelhamsteryt@gmail.com>, relay=127.0.0.1[127.0.0.1]:10024, delay=1.7, delays=0.13/0.01/0.01/1.6, dsn=2.0.0, status=sent (250 2.0.0 from MTA(smtp:[127.0.0.1]:10025): 250 2.0.0 Ok: queued as E33D722F63)
mail                 | Sep 15 18:35:27 mail postfix/qmgr[985]: 3F8FA227F4: removed
mail                 | Sep 15 18:35:27 mail dovecot: lmtp(1021): Connect from local
mail                 | Sep 15 18:35:27 mail dovecot: auth: ldap(pixelhamsteryt@gmail.com): unknown user
mail                 | Sep 15 18:35:27 mail postfix/lmtp[1020]: E33D722F63: to=<pixelhamsteryt@gmail.com>, relay=mail.melijn.com[/var/run/dovecot/lmtp], delay=0.06, delays=0.03/0.02/0.01/0.01, dsn=5.1.1, status=bounced (host mail.melijn.com[/var/run/dovecot/lmtp] said: 550 5.1.1 <pixelhamsteryt@gmail.com> User doesn't exist: pixelhamsteryt@gmail.com (in reply to RCPT TO command))
mail                 | Sep 15 18:35:27 mail dovecot: lmtp(1021): Disconnect from local: Logged out (state=READY)
mail                 | Sep 15 18:35:27 mail postfix/cleanup[1019]: F0B9522F6C: message-id=<20230915183527.F0B9522F6C@mail.melijn.com>
mail                 | Sep 15 18:35:27 mail postfix/qmgr[985]: F0B9522F6C: from=<>, size=3132, nrcpt=1 (queue active)
mail                 | Sep 15 18:35:27 mail postfix/bounce[1022]: E33D722F63: sender non-delivery notification: F0B9522F6C
mail                 | Sep 15 18:35:27 mail postfix/qmgr[985]: E33D722F63: removed
mail                 | Sep 15 18:35:27 mail dovecot: lmtp(1021): Connect from local
mail                 | Sep 15 18:35:28 mail dovecot: lmtp(merlijn@melijn.com)<1021><aIdIO++jBGX9AwAADnb+Ww>: sieve: msgid=<20230915183527.F0B9522F6C@mail.melijn.com>: stored mail into mailbox 'INBOX'
mail                 | Sep 15 18:35:28 mail postfix/lmtp[1020]: F0B9522F6C: to=<merlijn@melijn.com>, relay=mail.melijn.com[/var/run/dovecot/lmtp], delay=0.02, delays=0.01/0/0/0.01, dsn=2.0.0, status=sent (250 2.0.0 <merlijn@melijn.com> aIdIO++jBGX9AwAADnb+Ww Saved)
mail                 | Sep 15 18:35:28 mail postfix/qmgr[985]: F0B9522F6C: removed
mail                 | Sep 15 18:35:28 mail dovecot: lmtp(1021): Disconnect from local: Logged out (state=READY)
```
```
6504a3ef.37b599b7 0x7fdf9cbfb700 conn=1217 op=0 STARTTLS
6504a3ef.37b726e9 0x7fdf9cbfb700 conn=1217 op=0 RESULT oid= err=0 qtime=0.000014 etime=0.000147 text=
6504a3ef.3848a535 0x7fdfaec9d700 conn=1217 fd=17 TLS established tls_ssf=256 ssf=256 tls_proto=TLSv1.3 tls_cipher=TLS_AES_256_GCM_SHA384
6504a3ef.3868ba21 0x7fdfae49c700 conn=1217 op=1 BIND dn="cn=admin,dc=melijn,dc=com" method=128
6504a3ef.386c6bdf 0x7fdfae49c700 conn=1217 op=1 BIND dn="cn=admin,dc=melijn,dc=com" mech=SIMPLE bind_ssf=0 ssf=256
6504a3ef.387063e8 0x7fdfae49c700 conn=1217 op=1 RESULT tag=97 err=0 qtime=0.000017 etime=0.000565 text=
6504a3ef.3872e2db 0x7fdf9d3fc700 connection_input: conn=1217 deferring operation: binding
6504a3ef.3875e04b 0x7fdfae49c700 conn=1217 op=2 SRCH base="dc=melijn,dc=com" scope=2 deref=0 filter="(&(objectClass=inetOrgPerson)(?mailAlias=pixelhamsteryt@gmail.com))"
6504a3ef.3877bc3d 0x7fdfae49c700 conn=1217 op=2 SRCH attr=mail
6504a3ef.3879b6d3 0x7fdfae49c700 conn=1217 op=2 SEARCH RESULT tag=101 err=0 qtime=0.000173 etime=0.000447 nentries=0 text=
6504a3ef.387f4772 0x7fdf9edfe700 conn=1217 op=3 SRCH base="dc=melijn,dc=com" scope=2 deref=0 filter="(&(objectClass=inetOrgPerson)(?mailGroupMember=pixelhamsteryt@gmail.com))"
6504a3ef.3880469f 0x7fdf9edfe700 conn=1217 op=3 SRCH attr=mail
6504a3ef.3882fee8 0x7fdf9edfe700 conn=1217 op=3 SEARCH RESULT tag=101 err=0 qtime=0.000018 etime=0.000278 nentries=0 text=
6504a3ef.3888889a 0x7fdfaec9d700 conn=1217 op=4 SRCH base="dc=melijn,dc=com" scope=2 deref=0 filter="(&(objectClass=inetOrgPerson)(?mailAlias=@gmail.com))"
6504a3ef.3888f4e5 0x7fdfaec9d700 conn=1217 op=4 SRCH attr=mail
6504a3ef.388b3c9e 0x7fdfaec9d700 conn=1217 op=4 SEARCH RESULT tag=101 err=0 qtime=0.000017 etime=0.000230 nentries=0 text=
6504a3ef.38900a9d 0x7fdf9d3fc700 conn=1217 op=5 SRCH base="dc=melijn,dc=com" scope=2 deref=0 filter="(&(objectClass=inetOrgPerson)(?mailGroupMember=@gmail.com))"
6504a3ef.3890982c 0x7fdf9d3fc700 conn=1217 op=5 SRCH attr=mail
6504a3ef.38919a66 0x7fdf9d3fc700 conn=1217 op=5 SEARCH RESULT tag=101 err=0 qtime=0.000015 etime=0.000138 nentries=0 text=
6504a3ef.38af333b 0x7fdf9cbfb700 conn=1215 op=12 SRCH base="dc=melijn,dc=com" scope=2 deref=0 filter="(&(objectClass=inetOrgPerson)(?mailAlias=gmail.com))"
6504a3ef.38b02066 0x7fdf9cbfb700 conn=1215 op=12 SRCH attr=mail
6504a3ef.38b2f6dc 0x7fdf9cbfb700 conn=1215 op=12 SEARCH RESULT tag=101 err=0 qtime=0.000015 etime=0.000283 nentries=0 text=
6504a3ef.38d233bf 0x7fdfae49c700 conn=1215 op=13 SRCH base="dc=melijn,dc=com" scope=2 deref=0 filter="(&(objectClass=inetOrgPerson)(?mailGroupMember=gmail.com))"
6504a3ef.38d67ec0 0x7fdfae49c700 conn=1215 op=13 SRCH attr=mail
6504a3ef.38d9c769 0x7fdfae49c700 conn=1215 op=13 SEARCH RESULT tag=101 err=0 qtime=0.000021 etime=0.000538 nentries=0 text=
6504a3ef.38dff81b 0x7fdf9edfe700 conn=1215 op=14 SRCH base="dc=melijn,dc=com" scope=2 deref=0 filter="(mail=*@gmail.com)"
6504a3ef.38e0bf31 0x7fdf9edfe700 conn=1215 op=14 SRCH attr=mail
6504a3ef.38e49388 0x7fdf9edfe700 conn=1215 op=14 SEARCH RESULT tag=101 err=0 qtime=0.000018 etime=0.000331 nentries=1 text=
6504a3ef.3a4586dc 0x7fdfaec9d700 conn=1214 op=8 SRCH base="dc=melijn,dc=com" scope=2 deref=0 filter="(&(objectClass=inetOrgPerson)(uid=pixelhamsteryt))"
6504a3ef.3a46f557 0x7fdfaec9d700 conn=1214 op=8 SRCH attr=uid gidNumber
6504a3ef.3a4afd7e 0x7fdfaec9d700 conn=1214 op=8 SEARCH RESULT tag=101 err=0 qtime=0.000030 etime=0.000429 nentries=0 text=
6504a3ef.3ad4ed41 0x7fdf9d3fc700 conn=1217 op=6 SRCH base="dc=melijn,dc=com" scope=2 deref=0 filter="(&(objectClass=inetOrgPerson)(?mailAlias=merlijn@melijn.com))"
6504a3ef.3ad5df8c 0x7fdf9d3fc700 conn=1217 op=6 SRCH attr=mail
```
```


### Improvements to this form?

_No response_

**Comment by polarathene** [comment]
**TL;DR:**

This is not supported well with LDAP. Your main issue to address is the domains query for Postfix.
- It looks like you've got the gmail address configured as a `mail` attribute to forward inbound mail delivered to an alias in DMS.
- As your query to LDAP successfully matches a `mail` attribute when Postfix asks if it manages `@gmail.com`, it attempts to deliver locally rather than send outbound to Gmail servers. You can resolve this as described in the "Workaround" section below.
- Analysis from your logs + config is covered earlier if you (_or anyone else landing here_) wants to understand that better.

---

## Analysis

Relevant logs:

```
postfix/submission/smtpd[1005]: 3F8FA227F4: client=113.168-65-87.adsl-dyn.isp.belgacom.be[87.65.168.113], sasl_method=PLAIN, sasl_username=merlijn@melijn.com
postfix/qmgr[985]: E33D722F63: from=<merlijn@melijn.com>, size=1083, nrcpt=1 (queue active)
amavis[1003]: (01003-01) Passed CLEAN {RelayedOpenRelay}, [87.65.168.113]:54144 [87.65.168.113] <merlijn@melijn.com> -> <pixelhamsteryt@gmail.com>, Queue-ID: 3F8FA227F4, Message-ID: <a1351c88-8e7b-4fb6-bca1-53b455f05475@melijn.com>, mail_id: h6HUMdGGE7aH, Hits: -0.202, size: 871, queued_as: E33D722F63, 1602 ms
postfix/smtp-amavis/smtp[1013]: 3F8FA227F4: to=<pixelhamsteryt@gmail.com>, relay=127.0.0.1[127.0.0.1]:10024, delay=1.7, delays=0.13/0.01/0.01/1.6, dsn=2.0.0, status=sent (250 2.0.0 from MTA(smtp:[127.0.0.1]:10025): 250 2.0.0 Ok: queued as E33D722F63)
dovecot: auth: ldap(pixelhamsteryt@gmail.com): unknown user
postfix/lmtp[1020]: E33D722F63: to=<pixelhamsteryt@gmail.com>, relay=mail.melijn.com[/var/run/dovecot/lmtp], delay=0.06, delays=0.03/0.02/0.01/0.01, dsn=5.1.1, status=bounced (host mail.melijn.com[/var/run/dovecot/lmtp] said: 550 5.1.1 <pixelhamsteryt@gmail.com> User doesn't exist: pixelhamsteryt@gmail.com (in reply to RCPT TO command))
```

- Mail was submitted over port 587 for the login `merlijn@melijn.com`
- Sender address is the same as login, recipient is `pixelhamsteryt@gmail.com`
- Dovecot queried for the account `pixelhamsteryt@gmail.com`, user not recognized and Postfix rejects delivery to DMS with bounce message.

This tells me that DMS believes it manages this address, or possibly `@gmail.com`.

> Env: [mail-server.env.txt](https://github.com/docker-mailserver/docker-mailserver/files/12622664/mail-server.env.txt)

```env
# e.g. `"(&(objectClass=PostfixBookMailAccount)(uniqueIdentifier=%n))"`
DOVECOT_USER_FILTER="(&(objectClass=inetOrgPerson)(uid=%n))"
DOVECOT_USER_ATTRS="=home=/var/mail/%{ldap:uid},=mail=maildir:~/Maildir,=uid=5000,gidNumber=gid"

DOVECOT_PASS_ATTRS="uid=user,userPassword=password"
# e.g. `"(&(objectClass=PostfixBookMailAccount)(uniqueIdentifier=%n))"`

DOVECOT_PASS_FILTER="(&(objectClass=inetOrgPerson)(uid=%n))" # %n is the local part

# => Specify how ldap should be asked for domains
LDAP_QUERY_FILTER_DOMAIN="(mail=*@%s)"

# => Specify how ldap should be asked for users
LDAP_QUERY_FILTER_USER="(&(objectClass=inetOrgPerson)(mail=%s))"

# => Specify how ldap should be asked for aliases
LDAP_QUERY_FILTER_ALIAS="(&(objectClass=inetOrgPerson)(mailAlias=%s))"

# => Specify how ldap should be asked for groups
LDAP_QUERY_FILTER_GROUP="(&(objectClass=inetOrgPerson)(mailGroupMember=%s))"

# => Specify how ldap should be asked for senders
LDAP_QUERY_FILTER_SENDERS="(&(objectClass=inetOrgPerson)(|(mail=%s)(mailAlias=%s)(mailGroupMember=%s)))"

SPOOF_PROTECTION=0
```

There are various LDAP queries being made, but one returns a result for that gmail address:

```
SEARCH RESULT tag=101 err=0 qtime=0.000018 etime=0.000331 nentries=1 text=
SRCH base="dc=melijn,dc=com" scope=2 deref=0 filter="(&(objectClass=inetOrgPerson)(uid=pixelhamsteryt))"
SRCH attr=uid gidNumber
```

This looks like a query Dovecot presumably made? (_`DOVECOT_USER_FILTER` configured for `(&(objectClass=inetOrgPerson)(uid=%n))`_)

**EDIT:** Actually sorry, I am still not too familiar with LDAP and I noticed 9 `SRCH base` lines but only 8 `SEARCH RESULT`, so the Dovecot query failed (_as the logs indicate for the auth lookup_), and the `SEARCH RESULT` would actually belong to this:

```
6504a3ef.38dff81b 0x7fdf9edfe700 conn=1215 op=14 SRCH base="dc=melijn,dc=com" scope=2 deref=0 filter="(mail=*@gmail.com)"
6504a3ef.38e0bf31 0x7fdf9edfe700 conn=1215 op=14 SRCH attr=mail
6504a3ef.38e49388 0x7fdf9edfe700 conn=1215 op=14 SEARCH RESULT tag=101 err=0 qtime=0.000018 etime=0.000331 nentries=1 text=
```

### Cause identified

The result was for `mail` attribute that has a `@gmail.com` suffix in the value, with a returned `mail` attribute value (_the  `pixelhamsteryt@gmail.com` address_). This would match your query filter for Postfix to know what mail domains it manages:

```env
# => Specify how ldap should be asked for domains
LDAP_QUERY_FILTER_DOMAIN="(mail=*@%s)"
```

Thus this is a misconfiguration on your end. You have configured DMS with LDAP stating that it is responsible for the account `pixelhamsteryt@gmail.com`, which implies it owns the domain `gmail.com`.

To clarify, DMS can send mail from any account by default (`SPOOF_PROTECTION=1` restricts this). Technically accounts are user login names, they don't have to represent an email address (_although that's an assumption I've seen made with DMS contributions, docs and configuration_).
- You can have `john.doe@example.com` send with any sender address (`FROM`) that you like, even if that's your gmail address (_but unless DMS is authorized to send on behalf of `gmail.com` that probably won't be permitted, instead you'd use Gmail as a relay host that you submit mail from_).
- For sending mail to a recipient (_as you attempted to do with the gmail address_), DMS must not believe it owns that mail domain. If you used `mailAlias=pixelhamsteryt@gmail.com`, it would resolve the alias and deliver to your `mail` address it was associated to. This obviously would make no sense since no inbound mail should be received by DMS for a `@gmail.com`.
- What I assume has happened is you configured a `@melijn.com` account as the `mailAlias` to `mail=pixelhamsteryt@gmail.com`, hence why it matched the misconfigured query. This would be equivalent to `setup alias add hello@melijn.com pixelhamsteryt@gmail.com` + `setup email add pixelhamsteryt@gmail.com` (_your `mail` attribute, where this adds an account and detects the `@gmail.com` as managed by DMS_), not what you want.

---

## Workaround

[This blogpost](http://web.archive.org/web/20220811212530/https://b.aklein.me/mail-forwarding-with-postfix-and-ldap) describes how [they updated the `postfix-book.schema` to support a `mailForwardingAddress` attribute](https://github.com/variablenix/ldap-mail-schema/commit/e271d5f85b32fef887f341babc2f9028a7da07cb) on the `PostfixBookMailForward` object class which you could use as the `result_attribute`.

> ```
> server_host = ldap://ldap.example.com/
> search_base = ou=Mail,dc=example,dc=com
> version = 3
> bind = no
> query_filter = (&(|(mailAlias=%s)(mail=%s))(objectClass=PostfixBookMailForward))
> result_attribute = mailForwardingAddress
> ```

As you can see the `query_filter` is for any object entry with that class that has a `mailAlias` or `mail` attribute with the queried address (_`%s` token from Postfix_).
- The problem is presently DMS doesn't provide an easy way for you to target `result_attribtue` just for `ldap-aliases.cf`, you'd have to provide this file locally with the [default config template](https://github.com/docker-mailserver/docker-mailserver/blob/86edaf9a8a3ad8d5b2bba3d9a6ad94b675ca7d1a/target/postfix/ldap-aliases.cf).
- In the upcoming v13 release [this file will no longer exist](https://github.com/docker-mailserver/docker-mailserver/pull/3524) and it'll be easier to make this change via file or ENV.

That seems like the proper way to approach it with LDAP, since `mailAlias` attribute could be defined multiple times for `mail` and `mail` should ideally only represent an address that DMS is responsible for managing (receiving / sending), not forwarding.

I updated our test suite OpenLDAP image and config recently, but it presently does not use a `postfix-book.schema` with `mailForwardingAddress`, we could probably update that and the LDAP tests to cover this scenario, but as you may be aware not everyone uses the `postfix-book.schema`. Since the LDAP configuration is not predictable, with v13 much of the defaults will require the user to properly configure their queries that is appropriate for them 😅 

### Alternative workaround

In your case, you could probably get away with just fixing the `ldap-domains.cf`:

https://github.com/docker-mailserver/docker-mailserver/blob/86edaf9a8a3ad8d5b2bba3d9a6ad94b675ca7d1a/target/scripts/startup/setup.d/ldap.sh#L66

You'd either configure a dummy query filter with ENV such as `(|)` or with `user-patches.sh` script added, you can adjust the configuration:

```bash
#! /bin/bash

# Remove LDAP domain query:
postconf 'virtual_mailbox_domains = /etc/postfix/vhost'

# If you have more than just `melijn.com` mail domains DMS should manage,
# Add them to this file too:
# echo 'example.com' >> /etc/postfix/vhost
```

In your case with LDAP there is a default presently that assumes the mail domain for your mail accounts is from your DMS FQDN (`hostname: mail.melijn.com`) which extracts out `melijn.com`. So you shouldn't need to add anything else.

https://github.com/docker-mailserver/docker-mailserver/blob/86edaf9a8a3ad8d5b2bba3d9a6ad94b675ca7d1a/target/scripts/helpers/postfix.sh#L68-L75

## Additional Info to be aware of

In `v12.1` and earlier versions for LDAP, another difference (_seems like a bad contribution that slipped through_) drops internal mail delivery to system accounts (_which for you would be `root@mail.melijn.com` for example_), thus any failures/notifications from these will now be bounced/rejected from removing `$myhostname` as shown below:

https://github.com/docker-mailserver/docker-mailserver/blob/86edaf9a8a3ad8d5b2bba3d9a6ad94b675ca7d1a/target/scripts/startup/setup.d/ldap.sh#L78

That may not be relevant to you, as this technically requires setting up a `postfix-aliases.cf` file (_different from `postfix-virtual.cf` that `setup alias add` manages, or `mailAlias`_) that would direct it to the postmaster address. Virtual aliases could be configured to accomplish the same though.

**Expect breaking changes for LDAP with v13** as I try to address the stale support.

**Comment by polarathene** [comment]
> Sending emails to mailboxes outside (e.g. @gmail.com does not).

> I did not see any mention of configuration to which addresses you can send to unless I misunderstood the docs: https://docker-mailserver.github.io/docker-mailserver/latest/config/advanced/auth-ldap/.

So just to re-iterate:
- You should have no issue sending to another external address that doesn't have a domain in the value of a `mail` attribute.
- Those docs referenced are in the process of being rewritten for v13 release.

**Comment by ToxicMushroom** [comment]
Thanks for the very detailed response <3,

I resolved my issue as in one of the workarounds you suggested changing:
```
-LDAP_QUERY_FILTER_DOMAIN="(mail=*@%s)"
+LDAP_QUERY_FILTER_DOMAIN="(|)"
``` 
From what I found `(|)` is always false so it will never match. I'm no entirely sure what.
Now I can send mail to users inside my DMS and users outside.

Your assumption about what I was trying to do was not entirely correct so this might clear things up.
I did not have the gmail address as alias or anything but one of my LDAP users did have a gmail address in their mail field which was matching the filter I assume.
(The gmail address for the LDAP user was intened since they dont have to have an email on my DMS, the ldap is used for authenticating on other services aswell).
Instead I was trying to send it to an address that was not linked to any users in my LDAP so it'd go to a testing gmail inbox. 

Also thanks for the heads up for v13!

**Comment by polarathene** [comment]
> I did not have the gmail address as alias or anything but one of my LDAP users did have a gmail address in their mail field which was matching the filter I assume.

Oh ok, well that'd do it yes. It did highlight an issue with using LDAP for forwarding mail though so thanks for the misunderstanding anyway 😄 

> (The gmail address for the LDAP user was intended since they don't have to have an email on my DMS, the LDAP is used for authenticating on other services aswell).

So that's more of a misconfiguration on your end then.

You need to let DMS know with the filters what are valid mail accounts. This doesn't apply to just the domain check filter issue you had in that case.

In the `postfix-book.schema` there is an attribute `mailEnabled` which can provide an easy way to filter the results further by which ones are enabled for DMS. You could use any other attribute or filter query that accomplishes the same distinction (_maybe something like filtering by `memberOf` if you have groups setup_).

For example, there's a good chance that if your `LDAP_QUERY_FILTER_USER` would match that gmail account then they can probably login to DMS as if they had an account and send mail from DMS (_**as anyone**, unless `SPOOF_PROTECTION=1` is enabled but this will still allow sending as their `mail` address by default which would make your server look bad pretending to be Gmail_).
